### PR TITLE
[PSCmdAssistant] New-AzVM and Vmss default image to Win2022AzureEdition when user explicitly sets SecurityType to Standard

### PR DESCRIPTION
**Pull request by Sandido**
## Description

design request: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1408
When the SecurityType is explicitly set to Standard, now defaulting the image to Win2022AzureEdition to let users migrate to TrustedLaunch securityType more easily in the future. 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by BethanyZhou** [comment]
@Sandido , please check new added code, many test cases failed due to ` Object reference not set to an instance of an object.`.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 1]
```diff
@@ -74,16 +74,30 @@ public override void ExecuteCmdlet()
...
+                                // if securityType is Standard explicitly.
+                                else if (this.VirtualMachineScaleSet.VirtualMachineProfile?.SecurityProfile?.SecurityType?.ToLower() == ConstantValues.StandardSecurityType
+                                    && this.VirtualMachineScaleSet.VirtualMachineProfile?.StorageProfile?.ImageReference == null
+                                    && this.VirtualMachineScaleSet.VirtualMachineProfile?.StorageProfile?.OsDisk == null)
+                                {
+                                    this.ImageName = ConstantValues.TrustedLaunchDefaultImageAlias;
+                                    /*
+                                    this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference = new ImageReference
+                                    {
+                                        Publisher = "MicrosoftWindowsServer",
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:94]: cleanupo this

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 2]
```diff
@@ -667,11 +672,20 @@ async Task SimpleParameterSetExecuteCmdlet(IAsyncCmdlet asyncCmdlet)
...
+                this.ImageName = ConstantValues.TrustedLaunchDefaultImageAlias;
+            }
+
             // API does not currently support Standard securityType value, so need to null it out here. 
             if (this.IsParameterBound(c => c.SecurityType)
-                && this.SecurityType != null
-                && this.SecurityType.ToString().ToLower() == ConstantValues.StandardSecurityType)
+                && this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType)
             {
                 this.SecurityType = null;
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:690]: this seems unintuitive. does the API still not support Standard security type? why is that? do they plan to update this?

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 3]
```diff
@@ -681,7 +695,6 @@ async Task SimpleParameterSetExecuteCmdlet(IAsyncCmdlet asyncCmdlet)
             {
                 if (this.SecurityType?.ToLower() == ConstantValues.TrustedLaunchSecurityType || this.SecurityType?.ToLower() == ConstantValues.ConfidentialVMSecurityType)
                 {
-                    this.SecurityType = this.SecurityType;
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:684]: why is this line being removed?  Do we want to assign if TrustLaunch or ConfidentialVM security type?

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 4]
```diff
@@ -781,9 +780,15 @@ async Task StrategyExecuteCmdletAsync(IAsyncCmdlet asyncCmdlet)
                     this.EnableVtpm = this.EnableVtpm ?? true;
                     this.EnableSecureBoot = this.EnableSecureBoot ?? true;
                 }
+                // Default in TL Image to ease later migration.
```
Comment (grizzlytheodore) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:783]: i see this is the only part updated for New-AzVM,  
does this code path apply to both Simple and Default parameter set ?

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 5]
```diff
@@ -5271,4 +5271,45 @@ function Test-VirtualMachineScaleSetSecurityTypeAndFlexDefaults
...
+        $password = Get-PasswordForVM;
+        $adminPassword = $password | ConvertTo-SecureString -AsPlainText -Force;
+        $cred = New-Object System.Management.Automation.PSCredential ($adminUsername, $adminPassword);
+
+        # Requirements for the TrustedLaunch default behavior.
+        $vmss = New-AzVmss -ResourceGroupName $rgname -Credential $cred -VMScaleSetName $vmssName -SecurityType $securityTypeST -DomainNameLabel $domainNameLabel1;
+        
+        Assert-AreEqual $vmss.OrchestrationMode "Flexible";
+        Assert-Null $vmss.SecurityProfile;
+        Assert-AreEqual $vmss.VirtualMachineProfile.StorageProfile.ImageReference.Sku "2022-datacenter-azure-edition";
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:5308]: do we want to check for image here?

Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:5308]: shouldnt need re-recording

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 6]
```diff
@@ -7488,3 +7488,48 @@ function Test-CapacityReservationSharingProfile
...
+        $disable = $false;
+        $enable = $true;
+
+        $vm = New-AzVM -ResourceGroupName $rgname -Name $vmname -Credential $cred -SecurityType $securityTypeST -DomainNameLabel $domainNameLabel; 
+
+        $vm = Get-AzVm -ResourceGroupName $rgname -Name $vmname;
+
+        # Validate
+        Assert-AreEqual $vm.StorageProfile.ImageReference.Sku $SKU;
+        Assert-Null $vm.SecurityProfile;
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:7528]: do we want to check for image here?

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 2 above)
Comment (Sandido) [reply src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:690]: The API does not support Standard. IT is a dummy value for the client to use to opt out of the defaulting TL logic.

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 7]
```diff
@@ -542,15 +541,20 @@ private async Task<ResourceConfig<VirtualMachineScaleSet>> SimpleParameterSetOrc
                 {
                     if (_cmdlet.SecurityType?.ToLower() == ConstantValues.TrustedLaunchSecurityType || _cmdlet.SecurityType?.ToLower() == ConstantValues.ConfidentialVMSecurityType)
                     {
-                        _cmdlet.SecurityType = _cmdlet.SecurityType;
```
[the remark this answers is not in the document]
Comment (Sandido) [reply src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:545]: yes, and it is logically unnecessary to do.

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 4 above)
Comment (Sandido) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:783]: The ask is for only this parameter set, I have not tested the other.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 6 above)
Comment (Sandido) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:7528]: THat's what the sku check does.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 5 above)
Comment (Sandido) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:5308]: I check for image when I check for the images sku.

**Comment by Sandido** [comment]
@YanaXu , please consider this for this breaking change release.

File: src/Compute/Compute/ChangeLog.md [hunk 8]
```diff
@@ -25,10 +25,13 @@
 * Added `Etag` property to PSVirtualMachine and PSVirtualMachineScaleSet objects.   
 * Added parameters `-IfMatch` and `-IfNoneMatch` to `Update-AzVM`, `Update-AzVmss`, `New-AzVm`, `New-AzVmss`, `New-AzVmConfig`, and `New-AzVmssConfig` cmdlets.
 * Resolved the bug with `New-AzVMConfig` for `-CommunityGalleryImageId` and `-SharedGalleryImageId` parameters.
+* [Breaking Change] Added ValidateNotNullOrEmpty for `-ResourceGroupName` and `-VMScaleSetName` parameters to `Get-AzVmss` cmdlet. [#20095]
+* [Breaking Change] Added new business logic to `New-AzVmss` and `New-AzVM` cmdlets. When the user explicitly sets the `SecurityType` to `Standard`, the Image alias defaults to `Win2022AzureEdition` to make future migrations to Trusted Launch easier.
 
 ## Version 7.3.0
 * Added cmdlet `Invoke-AzSpotPlacementRecommender`.
 * Fixed `Update-AzCapacityReservationGroup` to remove Subscriptions from SharingProfile.
+* Added new optional parameter `SecureVMGuestStateSAS` to cmdlet `Grant-AzDiskAccess`.
```
Comment (YanaXu) [inline src/Compute/Compute/ChangeLog.md:34]: Why to add this line in "version 7.3.0". Please refer to https://www.powershellgallery.com/packages/Az.Compute/7.3.0. Usually we do not update the former versions. It doesn't work.

**Review by YanaXu — APPROVED** [review]

File: src/Compute/Compute/ChangeLog.md (hunk 8 above)
Comment (YanaXu) [reply src/Compute/Compute/ChangeLog.md:34]: Hi @Sandido , I've merged this PR to save time. Please double confirm why to add this line in a former release. If there is a reason, please comment in this PR and let's see what's the right thing to do.

Comment (Sandido) [reply src/Compute/Compute/ChangeLog.md:34]: @YanaXu , looks like that incorrectly merged. It should go in the latesdt release as that PR was merged on 4/25, https://github.com/Azure/azure-powershell/pull/24635

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 3 above)
Comment (Sandido) [reply src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:684]: It's logically unnecessary.

**Pull request merged**